internal/e2e/token_auth_test.go

```diff
@@ -0,0 +1,192 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e_test
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.wdf.sap.corp/cc/heureka/internal/api/graphql/access"
+	"github.wdf.sap.corp/cc/heureka/internal/util"
+	util2 "github.wdf.sap.corp/cc/heureka/pkg/util"
+
+	"github.wdf.sap.corp/cc/heureka/internal/server"
+
+	"github.com/golang-jwt/jwt/v5"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func sendGetRequest(url string, headers map[string]string) *http.Response {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", url, nil)
+	Expect(err).To(BeNil())
+	for k, v := range headers {
+		req.Header.Add(k, v)
+	}
+	resp, err := client.Do(req)
+	Expect(err).To(BeNil())
+	return resp
+}
+
+func unmarshalResponseData(resp *http.Response, respData interface{}) {
+	body, err := io.ReadAll(resp.Body)
+	Expect(err).To(BeNil())
+	err = json.Unmarshal(body, respData)
+	Expect(err).To(BeNil())
+}
+
+func generateJwtSpec(signingMethod jwt.SigningMethod, signKey interface{}, expiresAt *jwt.NumericDate) string {
+	claims := access.TokenClaims{
+		Version: "0.3.1",
+		RegisteredClaims: jwt.RegisteredClaims{
+			ExpiresAt: expiresAt,
+			IssuedAt:  jwt.NewNumericDate(time.Now()),
+			Issuer:    "heureka",
+			Subject:   "testUser",
+		},
+	}
+	token := jwt.NewWithClaims(signingMethod, claims)
+
+	tokenString, err := token.SignedString(signKey)
+	Expect(err).To(BeNil())
+	return tokenString
+}
+
+func generateJwt(jwtSecret string, expiresIn time.Duration) string {
+	expiresAt := jwt.NewNumericDate(time.Now().Add(expiresIn))
+	return generateJwtSpec(jwt.SigningMethodHS256, []byte(jwtSecret), expiresAt)
+}
+
+func generateInvalidJwt(jwtSecret string) string {
+	return generateJwtSpec(jwt.SigningMethodHS256, []byte(jwtSecret), nil)
+}
+
+func generateRsaPrivateKey() *rsa.PrivateKey {
+	privateKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	Expect(err).To(BeNil())
+	return privateKey
+}
+
+func generateJwtWithInvalidSigningMethod(jwtSecret string, expiresIn time.Duration) string {
+	privateKey := generateRsaPrivateKey()
+	expiresAt := jwt.NewNumericDate(time.Now().Add(expiresIn))
+	return generateJwtSpec(jwt.SigningMethodRS256, privateKey, expiresAt)
+}
+
+var _ = Describe("Getting access via API", Label("e2e", "TokenAuthorization"), func() {
```

Comment:
The tests fail on my machine: 

```shell
Running Suite: e2e Suite - /Users/Ixxxxx/repos/heureka/internal/e2e
====================================================================
Random Seed: 1725866514

Will run 7 of 135 specs
time="2024-09-09T09:21:58+02:00" level=info msg="Testing Connection to Database..."
time="2024-09-09T09:21:59+02:00" level=info msg="Testing Connection to Database..."
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSStime="2024-09-09T09:21:59+02:00" level=fatal msg="Error 29 (HY000): File './heurekanqmumbam4g/' not found (Errcode: 2 \"No such file or directory\")" error="Error 29 (HY000): File './heurekanqmumbam4g/' not found (Errcode: 2 \"No such file or directory\")"

Ginkgo ran 1 suite in 5.027070166s

Test Suite Failed

```

Reply:
This error seems not to be related to my change.

Reply:
That's true. Maybe @drochow can have a look as I was not able to find the root cause.

Reply:
Looks like some mess up with your DB data folder: 

- Delete local folder `<project_root>/.db`

Idk though , what is causing the mess up

Reply:
Seems fixed now. thx!